internal/datacoord/segment_manager.go

```diff
@@ -358,6 +387,12 @@ func (s *SegmentManager) DropSegment(ctx context.Context, segmentID UniqueID) {
 	defer sp.Finish()
 	s.mu.Lock()
 	defer s.mu.Unlock()
+	stat, ok := s.stats[segmentID]
+	if ok && stat != nil {
+		for _, allocation := range stat.allocations {
+			s.putAllocation(allocation)
```

Comment:
Would it be better if we use acquireAllocation&releaseAllocation  to name functions?